pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -350,12 +352,8 @@ func (c *Controller) MCSServices() []model.MCSServiceInfo {
 		mcsService.ClusterSetVIP = si.clusterSetVIP
 	}
 
-	out := make([]model.MCSServiceInfo, 0, len(outMap))
-	for _, v := range outMap {
-		out = append(out, *v)
-	}
-
-	return out
+	out := maps.Values(outMap)
+	return slices.Dereference(out)
```

Comment:
why not return out

Reply:
should return []*model.MCSServiceInfo) not []model.MCSServiceInfo, slices.Dereference() can do it

Reply:
can we change outMap to []model.MCSServiceInfo, then we donot need to convert any more

Reply:
Thank review, I simplified it with your suggestion